lib/gitsh/tab_completion/README.md

```diff
@@ -71,6 +72,11 @@ implementation has two differences:
    generating behaviour, e.g. if we're expecting a file system path we'd match
    on any input, but only generate valid paths to files that really exist.
 
+3. We introduce the concept of a _fallback transition_, which is a transition
```

Comment:
if you use only `1.` for the markdown list, then you won't have to worry about incrementing the numbers in the future

Reply:
I like that markdown reads nicely in a plain-text context as well as when rendered, for example I often read a repo's README with vim or less. I think I'm OK with the diff noise of changing numbers to make things more readable when they're not compiled to HTML and rendered by a browser.